custom/zipline/api.py

```diff
@@ -0,0 +1,124 @@
+from custom.zipline.models import EmergencyOrder, EmergencyOrderStatusUpdate
+from datetime import datetime
+from decimal import Decimal
+from dimagi.utils.couch import CriticalSection
+
+
+class ProductQuantity(object):
+    """
+    A simple class to describe a product and the quantity of it to be ordered.
+    """
+
+    def __init__(self, code, quantity):
+        """
+        :param code: the product code of the product being requested
+        :param quantity: the quantity of the product being requested
+        """
+        self.code = code
+        self.quantity = quantity
+
+
+def initiate_emergency_order(domain, user, phone_number, location, products):
+    """
+    :param domain: the domain in which the order is being requested
+    :param user: the user who is initiating the emergency order request
+    :param phone_number: the phone_number (string) of the user who is initiating the emergency order request
+    :param location: the SQLLocation where the products should be delivered
+    :param products: a list of ProductQuantity objects representing the products to be ordered
+    """
+    from custom.zipline.tasks import send_emergency_order_request
+
+    order = EmergencyOrder.objects.create(
+        domain=domain,
+        requesting_user_id=user.get_id,
+        requesting_phone_number=phone_number,
+        location=location,
+        location_code=location.site_code,
+        products_requested=[{'code': p.code, 'quantity': p.quantity} for p in products],
+        timestamp=datetime.utcnow()
+    )
+
+    send_emergency_order_request(order.pk)
+
+
+def process_receipt_confirmation(domain, user, location, products):
+    """
+    Processes a receipt confirmation of delivered commodities.
+    :param domain: the domain
+    :param user: the user that sent the receipt confirmation
+    :param location: the SQLLocation that the confirmation is coming from
+    :param products: a list of ProductQuantity objects representing the products
+    and quantities that are being reported to have been received
+    """
+    order = find_order_to_apply_confirmation(domain, location)
```

Comment:
It looks like the request body will have an `orderId`, should that be used to identify the order?